tests/manifests/test_serviceaccounts.py

```diff
@@ -104,6 +104,8 @@ async def test_does_not_create_serviceaccounts_if_configured_not_to(values, make
         serviceaccount_names = set()
         covered_serviceaccount_names = set()
         for template in await make_templates(values_to_modify):
+            if template_to_deployable_details(template) != deployable_details:
```

Comment:
We are re-generating all templates when disabling the service account for only 1 deployable details. This caused issues such as the test detected a ServiceAccount not related to the currently tested deployable.

```
E                   AssertionError: Job/pytest-phwgyajqijitsjyhgvdeke-synapse-check-config (for synapse) specified an existing ServiceAccount: pytest-phwgyajqijitsjyhgvdeke-synapse-check-config
```